### DISCUSSION TITLE
Audiogridder functionality

### DISCUSSION THREAD

**Discussion by williamcorney** [apohl79/audiogridder#1262] [Q&A]
Could someone clarify my understanding of audiogridder?

if I was to to run a VST in just wine, I get dreadful latency, which I understand to because it’s using direct sound and  the bulk of audio processing is being done effectively in the wine session.

However using audio gridder we are ‘offloading the DSP processing of audio plugins to remote computers such that the DSP code of the loaded plugins will be executed on the server machine’ 

Does this mean in plain English that the bulk of the audio processing is being done by the client pc connected to the plug-in?  In my case this is macOS ( and the audio gridder server runs inside wine on same hardware) and the VST plugin is interfacing just with core audio , rather than going via direct sound.  The reason why I get much better performance ( and very low latency ) has to do with the VST plugin can directly talk to core audio ( rather than some convoluted path via direct audio ).  The performance I am getting therefore would be expected to be almost as good as if the plug-in were a native plug-in running on macOS ?

**Comment by apohl79** [comment]
The processing happens in the server process which is hosting the VST in your case. If you use AG to run the server on a separate computer (the original use case) you are offloading the audio processing, as the statement is suggesting. But in your case running it on the same machine, there is no offloading of course.

**Comment by williamcorney** [comment]
I’ll clarify what I mean

if I run VST in wine without audiogridder, my understanding is that the processing is all occurring within wine and the output goes to my mac through direct sound and I get latency 

eg VST > WINE > DIRECT SOUND > CORE AUDIO ( latency )

If I run VST in wine with audio gridder my understanding is that audiogridder is passing the processing to the main OS ie my mac , and this is why I don’t get latency?

eg VST > WINE > AUDIOGRIDDER SERVER > AUDIOGRIDDER CLIENT > CORE AUDIO ( no latency )

my understanding is this tunnel for want of a better word , means the VST functions as well in the mac as if were a native mac VST ( when in fact it’s actually a windows VST running in a wine instance ).  

If I’ve got the completely wrong end of the stick, what you are saying suggests ALL the processing is being done inside the wine session ( which is running the audiogridder server ) and this is then being passed through to my Mac OS .
